cmdx.py

```diff
@@ -3917,6 +3914,37 @@ def parent(self, node, parent=None):
         create_node = createNode
 
 
+class DGContext(om.MDGContext):
+
+    def __init__(self, time=None):
+        """Context for evaluating the Maya DG
+
+        Extension of MDGContext to also accept time as a float. In Maya 2018
+        and above DGContext can also be used as a context manager.
+
+        Arguments:
+            time (float, om.MTime, optional): Time at which to evaluate context
+
+        """
+
+        if time is not None:
+            if isinstance(time, (int, float)):
+                time = om.MTime(time, om.MTime.uiUnit())
+            super(DGContext, self).__init__(time)
+        else:
+            super(DGContext, self).__init__()
+        self._previousContext = None
+
+    if __maya_version__ >= 2018:
```

Comment:
Hm, will this not break any call to..

```python
with cmdx.DGContext(50):
   # ...
```

If the same code is run on Maya 2017?

An alternative might be to keep the `__enter__` and `__exit__` but to call `cmds.warning("Sorry dude, this no work")` or something like that. Or is there `cmds.error` too? Something loud! :D

Reply:
Yeah that's correct. I've just pushed an update, is that what you had in mind?

Reply:
Yes, that's it. Thanks